### ISSUE TITLE
Received "user canceled the authentication" without canceling it

### ISSUE THREAD

**Issue by SuperJMN** [AzureAD/azure-activedirectory-library-for-dotnet#684]
I just called:

```
            var authResult = await context.AcquireTokenAsync(resource, clientId, uri, PromptBehavior.Auto);
```

And instead of showing the window requesting credentials, it threw and exception as if I had dismissed a window (that never appeared).

What's happening? :(

**Comment by jmprieur** [comment]
@SuperJMN : can you tell us a bit more about your scenario? share some repro?

**Comment by kpanwar** [comment]
This is for UWP app, right?

**Comment by dinobud** [comment]
My users and I are also experiencing this issue but we all see a brief flash of the authentication dialog. I do have one machine on which the dialog appeared correctly and perhaps coincidentally has had a clean install of Windows 10 Creators update a few days prior. My app is WPF net462. If I can help in any way, let me know.

**Comment by jmprieur** [comment]
which version of ADAL.Net are you using  @dinobud ? do you have repro code that we could investigate?

**Comment by kpanwar** [comment]
@SuperJMN - this could happen if there is a HTTP redirect at anytime during authentication.

**Comment by jmprieur** [comment]
@SuperJMN : in case (like me) you did not understand @kpanwar's answer, what he means is that this error happens when authentication flow is redirected through **http** instead of **https** in any part of the flow. We advise you not to use http servers.

**Comment by jennyf19** [comment]
Reopening this issue - possibly connected to [this issue in azure-samples](https://github.com/Azure-Samples/active-directory-dotnet-native-multitarget/issues/17) and this [issue in ADAL](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/issues/969) - @SomkaPe is working on a fix.

**Comment by henrik-me** [comment]
#969 is closed. @jennyf19 anything else we need to verify?

**Comment by jennyf19** [comment]
@henrik-me I don't believe so...the customer verified it, as did we. you can see the notes in the attached [issue](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/issues/969).

**Comment by jennyf19** [comment]
Addressed in [3.19.2](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/releases/tag/v3.19.2)
